netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -94,6 +95,8 @@
   private Channel channel;
   /** Since not thread-safe, may only be used from event loop. */
   private ClientTransportLifecycleManager lifecycleManager;
+  private Attributes attributes = Attributes.EMPTY;
```

Comment:
As the API is currently designed, I think it would need to be. However, I'd suggest leaving this non-volatile and in `ClientCall.getAttributes()` saying that it is only defined after receiving response headers/trailers.